### PR TITLE
Add a note on setMatching Behavior and fix listing

### PR DESCRIPTION
**Pull request by antograssiot**
refers to cakephp/cakephp#10000

### CODE REVIEW

**Comment by markstory** [comment]
Other than the conflic this looks good.

**Comment by dereuromark** [comment]
I put those in backticks on 3.next so they are in sync with other namespace statements on the page.

**Comment by antograssiot** [comment]
Derp I will fix it

**Comment by antograssiot** [comment]
ok I fixed the conflicts.
No problem @dereuromark, I didn't notice your PR, it would have avoid me extra work twice :(
Great if it looks good now 👍

File: en/appendices/3-4-migration-guide.rst [hunk 1]
```diff
@@ -156,7 +156,8 @@ The following is a list of methods that are deprecated and replaced with
     * ``config()``
     * setter part of ``canBeJoined()`` (now ``setCanBeJoined()``)
 ``Cake\ORM\EagerLoader``
-    * ``matching()``
+    * ``matching()`` (``getMatching()`` will have to be call after ``setMatching()`` to
```
Comment (HavokInspiration) [inline en/appendices/3-4-migration-guide.rst]: Shouldn't it read "will have to be call**ED**" ?

Comment (HavokInspiration) [inline en/appendices/3-4-migration-guide.rst]: I updated #4593 with this, any native English speaker can correct me if I'm wrong :)

**Pull request merged**